### PR TITLE
Feature/edit workspace settings

### PR DESCRIPTION
**Pull request by tomsmith8**
# Add Workspace Editing Feature to Settings Page

This PR implements a comprehensive workspace editing feature that allows workspace owners and admins to update workspace name, slug, and description directly from the settings page.

 **Features Added**

  - Workspace Profile Editing: Clean, intuitive form for updating workspace details
  - Smart URL Handling: Automatic redirection when workspace slug changes
  - Permission Controls: Restricted to OWNER and ADMIN roles only
  - Real-time Validation: Comprehensive client and server-side validation
  - User Feedback: Toast notifications and loading states for better UX

##  **Architecture & Implementation**

  **Backend Changes:**
  - PUT /api/workspaces/[slug] endpoint with full error handling
  - updateWorkspace service function with permission checks
  - Zod validation schema with existing workspace constants
  - Proper database constraint handling and conflict resolution

 **Frontend Changes:**
  - WorkspaceSettings component with React Hook Form integration
  - Optimistic UI updates and form state management
  - Integration into existing Settings page layout
  - Proper TypeScript interfaces and error handling

 **Key Technical Decisions:**
  - Used PUT method (full resource update) over PATCH
  - Leveraged existing useWorkspaceAccess hook for permissions
  - Implemented slug change detection for URL redirection
  - Maintained consistency with existing codebase patterns

 **Test Coverage**

  Added 33 comprehensive tests covering all layers:
  - 15 validation schema tests - edge cases, error messages, data transformation
  - 8 service function tests - permissions, business logic, error scenarios
  - 10 API integration tests - HTTP methods, auth, validation, responses

 **Files Changed**

  - 9 files modified, 1,237 lines added
  - Core implementation: types, schemas, services, API routes, UI component
  - Comprehensive test suite across unit and integration levels
  - Zero breaking changes to existing functionality

 **Security & Validation**

  - Role-based access control (OWNER/ADMIN only)
  - Input sanitization and validation at multiple layers
  - Slug uniqueness checks and reserved word protection
  - Proper error handling without information disclosure

### CODE REVIEW

File: src/components/WorkspaceSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      slug: workspace?.slug || "",
+      description: workspace?.description || "",
+    },
+  });
+
+  const onSubmit = async (data: UpdateWorkspaceInput) => {
+    if (!workspace) return;
+
+    setIsSubmitting(true);
+    try {
```
Comment (Rassl) [inline src/components/WorkspaceSettings.tsx:54]: blank line

**Review by Rassl — APPROVED** [review]
LGTM

**Comment by fayekelmith** [comment]
> @fayekelmith review would be good thanks!

looks good to me, sir. LGTM

File: src/components/WorkspaceSettings.tsx (hunk 1 above)
Comment (tomsmith8) [reply src/components/WorkspaceSettings.tsx:54]: fixed
